main.nf

```diff
@@ -7,96 +7,133 @@ nextflow.enable.dsl=2
-         def out_dir = "${params.output_dir}/${repo_name}"
-         tuple(repo_url, repo_name, out_dir)
+
+    // Load .env file
+    loadEnvFile('.env')
+
+    // Parameter validation
+    if (!params.repo_url && !params.sample_sheet) {
+        throw new IllegalArgumentException("ERROR: Provide either a sample_sheet or repo_url parameter")
+    }
+
+    if (params.upload_to_synapse && !params.synapse_folder_id) {
+        throw new IllegalArgumentException("ERROR: synapse_folder_id must be provided when --upload_to_synapse is true.")
+    }
+
+    if (!params.synapse_agent_id) {
+        throw new IllegalArgumentException("ERROR: synapse_agent_id must be provided.")
+    }
+
+    // Validate repository URL format
+    def validateRepoUrl = { url ->
+        if (!url) return false
+        def validUrlPattern = ~/^https:\/\/github\.com\/[^\/]+\/[^\/]+\.git$/
+        return url ==~ validUrlPattern
+    }
+
+    // Extract repository name from URL
+    def getRepoName = { url ->
+        def urlStr = url instanceof List ? url[0] : url
+        return urlStr.tokenize('/')[-1].replace('.git','')
     }
-    
-    // Process each repository with ProcessRepo (clones repo and performs initial checks)
-    def repoOutputs = repoTuples | ProcessRepo
-    
-    // Run the Almanack analysis on each repository
-    def almanackOutputs = repoOutputs | RunAlmanack
-
-    // Collect all unique status files into one list
-    almanackOutputs
-        .map { repo_url, repo_name, out_dir, status_file -> status_file }
-        .collect()
-        .set { allStatusFiles }
-    
-    // Generate the consolidated report from all status files
-    allStatusFiles | GenerateReport
+
+    // Create a channel of repo URLs
+    Channel.from(
+        params.sample_sheet ?
+            file(params.sample_sheet).readLines().drop(1).collect { it.trim() }.findAll { it } :
+            [params.repo_url]
+    ).set { repo_urls }
+
+    // Validate and process each repo
+    repo_urls.map { repo_url ->
+        if (!validateRepoUrl(repo_url)) {
+            throw new IllegalArgumentException("ERROR: Invalid repository URL format. Expected: https://github.com/username/repo.git")
+        }
+        def repo_name = getRepoName(repo_url)
+        tuple(repo_url, repo_name, params.output_dir)
+    }.set { repo_tuples }
+
+    // Process repository
+    ProcessRepo(repo_tuples)
+
+    // Run Almanack
+    RunAlmanack(ProcessRepo.out)
+
+    // Execute tests
+    TestExecutor(ProcessRepo.out)
+
+    // Combine outputs for JOSS analysis
+    ProcessRepo.out
+        .combine(RunAlmanack.out, by: [0,1])
+        .combine(TestExecutor.out, by: [0,1])
+        .map { it ->
+            tuple(
+                it[0],   // repo_url
+                it[1],   // repo_name
+                it[2],   // repo_dir
+                it[3],   // out_dir
+                it[4],   // status_file
+                it[7],   // almanack_results
+                it[8]    // test_results
```

Comment:
Can you grab these by name kinda like this? https://github.com/Sage-Bionetworks-Workflows/nf-synapse-challenge/blob/9309c7d67e33fdd144dd8282c71b495be7695001/workflows/DATA_TO_MODEL.nf#L69